client/src/AddVideo.js

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from "react";
+
+export default function AddVideo({ addVideo }) {
+  let [input, setInput] = useState({
+    title: "",
+    url: "",
+    rating: 0,
+  });
+
+  const handleChange = (e) => {
+    setInput({ ...input, [e.target.name]: e.target.value });
+  };
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    input.id = uuidv4();
+    addVideo(input);
+  };
+
+  function uuidv4() {
```

Comment:
maybe you can use the uuid library for this unless there is a valid reason to build your own random character generator 

https://www.npmjs.com/package/uuid if you are using npm
https://yarnpkg.com/package/uuidv4 uf you are using yarn